#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.08129940113128242 0.9926936358605647 0.089161385687007 4.546551044209172
  position -2.233471410498927 1.1461308025283365 -0.15843028511894397
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 100 100
  floorTileSize 1 1
  wallHeight 0.05
}
Robot {
  translation -0.656517 0.04 1.03568
  rotation 0 1 0 1.8326
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0 0 1 1.57
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF Physcis_wh Physics {
        }
      }
    }
    SliderJoint {
      jointParameters JointParameters {
        axis 0 1 0
      }
      device [
        LinearMotor {
          name "linear"
          maxPosition 0.2
        }
      ]
      endPoint Solid {
        translation -0.085 0.05 0
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 0 1
              metalness 0
            }
            geometry Box {
              size 0.15 0.02 0.02
            }
          }
        ]
        name "solid(5)"
      }
    }
    Solid {
      translation 0 0.15 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.2 0.5 0.4
            metalness 0
          }
          geometry Box {
            size 0.02 0.25 0.02
          }
        }
      ]
      name "solid(4)"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 0 0 1 1.57
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics USE Physcis_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 0 0 1 1.57
        children [
          USE Wheel
        ]
        name "solid(2)"
        boundingObject USE Wheel
        physics USE Physcis_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 0 0 1 1.57
        children [
          USE Wheel
        ]
        name "solid(3)"
        boundingObject USE Wheel
        physics USE Physcis_wh
      }
    }
    DistanceSensor {
      translation 0.02 0 0.1
      rotation 0 1 0 -1.27
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0 0.1
      rotation 0 1 0 -1.87
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
  ]
  boundingObject USE Body
  physics Physics {
  }
  controller "Linear_actuator"
}
Solid {
  translation 0.942831 0.2 -0.630653
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Sphere {
        radius 0.1
      }
    }
  ]
  boundingObject Sphere {
    radius 0.1
  }
  physics Physics {
  }
}
Solid {
  translation -0.0697707 0.2 1.30985
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Sphere {
        radius 0.1
      }
    }
  ]
  name "solid(1)"
  boundingObject Sphere {
    radius 0.1
  }
  physics Physics {
  }
}
Robot {
  translation 0 0.491 0
  children [
    DEF low_spine Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0
        metalness 0
      }
      geometry Capsule {
        height 0.144
        radius 0.03
      }
    }
    BallJoint {
      jointParameters BallJointParameters {
      }
      jointParameters2 JointParameters {
      }
      jointParameters3 JointParameters {
      }
      device [
        RotationalMotor {
        }
      ]
      device2 [
        RotationalMotor {
          minPosition -1.5707963267948966
          maxPosition 1.5707963267948966
        }
      ]
      device3 [
        RotationalMotor {
        }
      ]
      endPoint Solid {
        translation 0 0.19 0
        children [
          DEF up_spine Shape {
            appearance PBRAppearance {
              metalness 0
            }
            geometry Capsule {
              height 0.144
              radius 0.02
            }
          }
        ]
        boundingObject USE up_spine
        physics Physics {
        }
      }
    }
    BallJoint {
      jointParameters BallJointParameters {
      }
      jointParameters2 JointParameters {
      }
      jointParameters3 JointParameters {
      }
      device [
        RotationalMotor {
        }
      ]
      device2 [
        RotationalMotor {
          minPosition -1.5707963267948966
          maxPosition 1.5707963267948966
        }
      ]
      device3 [
        RotationalMotor {
        }
      ]
      endPoint Solid {
        translation 0 0.278 0.139
        rotation 1 0 0 1.57
        children [
          DEF shoulder Shape {
            appearance PBRAppearance {
              metalness 0
            }
            geometry Capsule {
              height 0.2193
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE shoulder
        physics Physics {
        }
      }
    }
    BallJoint {
      jointParameters BallJointParameters {
      }
      jointParameters2 JointParameters {
      }
      jointParameters3 JointParameters {
      }
      device [
        RotationalMotor {
        }
      ]
      device2 [
        RotationalMotor {
          minPosition -1.5707963267948966
          maxPosition 1.5707963267948966
        }
      ]
      device3 [
        RotationalMotor {
        }
      ]
      endPoint Solid {
        translation 0 0.278 -0.139
        rotation 1 0 0 1.57
        children [
          DEF shoulder Shape {
            appearance PBRAppearance {
              metalness 0
            }
            geometry Capsule {
              height 0.2193
              radius 0.02
            }
          }
        ]
        name "solid(2)"
        boundingObject USE shoulder
        physics Physics {
        }
      }
    }
    BallJoint {
      jointParameters BallJointParameters {
      }
      jointParameters2 JointParameters {
      }
      jointParameters3 JointParameters {
      }
      device [
        RotationalMotor {
        }
      ]
      device2 [
        RotationalMotor {
          minPosition -1.5707963267948966
          maxPosition 1.5707963267948966
        }
      ]
      device3 [
        RotationalMotor {
        }
      ]
      endPoint Solid {
        translation 0 -0.092 -0.109
        rotation 1 0 0 1.57
        children [
          DEF pelvis Shape {
            appearance PBRAppearance {
              metalness 0
            }
            geometry Capsule {
              height 0.1615
              radius 0.02
            }
          }
        ]
        name "solid(3)"
        boundingObject USE pelvis
        physics Physics {
        }
      }
    }
    BallJoint {
      jointParameters BallJointParameters {
      }
      jointParameters2 JointParameters {
      }
      jointParameters3 JointParameters {
      }
      device [
        RotationalMotor {
        }
      ]
      device2 [
        RotationalMotor {
          minPosition -1.5707963267948966
          maxPosition 1.5707963267948966
        }
      ]
      device3 [
        RotationalMotor {
        }
      ]
      endPoint Solid {
        translation 0 -0.092 0.109
        rotation 1 0 0 1.57
        children [
          DEF pelvis Shape {
            appearance PBRAppearance {
              metalness 0
            }
            geometry Capsule {
              height 0.1615
              radius 0.02
            }
          }
        ]
        name "solid(4)"
        boundingObject USE pelvis
        physics Physics {
        }
      }
    }
  ]
  name "human_robot"
  boundingObject USE low_spine
  physics Physics {
  }
}
